osf_tests/test_guid.py

```diff
@@ -161,6 +161,51 @@ def test_resolve_guid_no_url(self):
         )
         assert res.status_code == 404
 
+    def test_resolve_guid_no_auth_redirect_to_cas_includes_public(self):
+        """
+        Unauthenticated users are sent to login when visiting private projects, but not if the projects are public.
+        """
+        res = self.app.get(
+            self.node.web_url_for('resolve_guid', guid=self.node._id),
```

Comment:
I feel like testing for just `files` was too specific to this particular project, but I realize the logic involving the different suffixes is necessary, so I'll just add another test that specifically tests for segments, this will help as future segments are Emberized.